### PR TITLE
Added explaination on context in events and initial marking

### PR DESCRIPTION
**Pull request by epitre**

### CODE REVIEW

**Comment by pluk77** [comment]
@epitre Nice solution, which will indeed also solve the issue of the non-dispatched initial entered event.

**Comment by pluk77** [comment]
@epitre Perhaps you can also add a note about the initial marking option to make it clear it is optional and will default to the first place if not defined.

**Comment by epitre** [comment]
> @epitre Perhaps you can also add a note about the initial marking option to make it clear it is optional and will default to the first place if not defined.

it is explained here : https://github.com/symfony/symfony-docs/pull/13947/files#diff-c94d89e3669d4257383cfe5faf73e644R97

**Comment by pluk77** [comment]
> 
> 
> > @epitre Perhaps you can also add a note about the initial marking option to make it clear it is optional and will default to the first place if not defined.
> 
> it is explained here : https://github.com/symfony/symfony-docs/pull/13947/files#diff-c94d89e3669d4257383cfe5faf73e644R97

I was more referring to the config option `initial_marking` in the workflow configuration file (yml, xml, php). It is not clear from the current docs that the initial marking will always be the place that is first in the list of defined places, unless you set a different place using `initial_marking`.

Even if you do not instantiate the workflow by using `getMarking()` but go directly to `apply()` the subject will still be placed in the initial place first if there is no marking on the subject yet. 

Hope I am making sense?

File: components/workflow.rst [hunk 1]
```diff
@@ -94,6 +94,16 @@ you can retrieve a workflow from it and use it as follows::
     $workflow->can($blogPost, 'publish'); // True
     $workflow->getEnabledTransitions($blogPost); // $blogPost can perform transition "publish" or "reject"
 
+If you want to initiate your worflow, you can simply call ``getMarking``::
```
Comment (u-mulder) [inline components/workflow.rst:97]: `workflow` with __k__

Comment (epitre) [reply components/workflow.rst]: thanks

**Comment by lyrixx** [comment]
FYI, The pull request has been merged

**Comment by nicolas-grekas** [comment]
This PR has been closed because the master has been removed.
Please submit it again against the appropriate branch.

**Comment by epitre** [comment]
@nicolas-grekas I thought it was merge as said by @lyrixx ?

Anyway, I made a new PR : https://github.com/symfony/symfony-docs/pull/14678

**Comment by nicolas-grekas** [comment]
I think @lyrixx meant that the feature has been merged on the code side (but not on the doc side.)

**Pull request closed without merging**